pkg/controller/pravega/pravega_controller.go

```diff
@@ -160,3 +163,30 @@ func MakeControllerService(p *api.PravegaCluster) *corev1.Service {
 		},
 	}
 }
+
+func MakeControllerPodDisruptionBudget(pravegaCluster *api.PravegaCluster) *policyv1beta1.PodDisruptionBudget {
+	var maxUnavailable intstr.IntOrString
+
+	if pravegaCluster.Spec.Pravega.ControllerReplicas == int32(1) {
+		maxUnavailable = intstr.FromInt(0)
+	} else {
+		maxUnavailable = intstr.FromInt(1)
+	}
+
+	return &policyv1beta1.PodDisruptionBudget{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PodDisruptionBudget",
+			APIVersion: "policy/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.PdbNameForController(pravegaCluster.Name),
+			Namespace: pravegaCluster.Namespace,
+		},
+		Spec: policyv1beta1.PodDisruptionBudgetSpec{
+			MaxUnavailable: &maxUnavailable,
```

Comment:
Could you elaborate on the reason why we are not using MinUnavailable for controller? (if we have multiple controllers on the same node, we can drain all the controllers provided there are instances of a controller running other nodes...)

Specifying MinUnavailable = 1 and removing MaxUnavailable(as they are mutually exclusive settings) we could remove `if else ` at line 170.

Reply:
You are right, being a stateless component, `minAvailable=1` makes more sense. Updated it.